crates/ruff/src/rules/flake8_type_checking/rules/runtime_import_in_type_checking_block.rs

```diff
@@ -61,72 +67,172 @@ impl Violation for RuntimeImportInTypeCheckingBlock {
+                RuntimeImportInTypeCheckingBlock {
+                    qualified_name: qualified_name.to_string(),
+                },
+                trimmed_range,
+            );
+            if let Some(range) = parent_range {
+                diagnostic.set_parent(range.start());
+            }
+            if let Some(fix) = fix.as_ref() {
+                diagnostic.set_fix(fix.clone());
+            }
+            diagnostics.push(diagnostic);
+        }
+    }
+
+    // Separately, generate a diagnostic for every _ignored_ import, to ensure that the
+    // suppression comments aren't marked as unused.
+    for Import {
+        qualified_name,
+        trimmed_range,
+        parent_range,
+        ..
+    } in ignores_by_statement.into_values().flatten()
     {
         let mut diagnostic = Diagnostic::new(
             RuntimeImportInTypeCheckingBlock {
                 qualified_name: qualified_name.to_string(),
             },
-            binding.trimmed_range(checker.semantic_model(), checker.locator),
+            trimmed_range,
         );
-        if let Some(range) = binding.parent_range(checker.semantic_model()) {
+        if let Some(range) = parent_range {
             diagnostic.set_parent(range.start());
         }
+        diagnostics.push(diagnostic);
+    }
+}
 
-        if checker.patch(diagnostic.kind.rule()) {
-            diagnostic.try_set_fix(|| {
-                // Step 1) Remove the import.
-                // SAFETY: All non-builtin bindings have a source.
-                let source = binding.source.unwrap();
-                let stmt = checker.semantic_model().stmts[source];
-                let parent = checker.semantic_model().stmts.parent(stmt);
-                let remove_import_edit = autofix::edits::remove_unused_imports(
-                    std::iter::once(qualified_name),
-                    stmt,
-                    parent,
-                    checker.locator,
-                    checker.indexer,
-                    checker.stylist,
-                )?;
-
-                // Step 2) Add the import to the top-level.
-                let reference = checker.semantic_model().references.resolve(*reference_id);
-                let add_import_edit = checker.importer.runtime_import_edit(
-                    &StmtImport {
-                        stmt,
-                        qualified_name,
-                    },
-                    reference.range().start(),
-                )?;
-
-                Ok(
-                    Fix::suggested_edits(remove_import_edit, add_import_edit.into_edits())
-                        .isolate(checker.isolation(parent)),
-                )
-            });
-        }
+/// A runtime-required import with its surrounding context.
+struct Import<'a> {
+    /// The qualified name of the import.
+    qualified_name: &'a str,
+    /// The first reference to the imported symbol.
+    reference_id: ReferenceId,
+    /// The trimmed range of the import.
```

Comment:
Is this like `List` in `from typing import List, Sequence`?

Reply:
Yeah, I'll add a note.